cmd/agent/common/common_windows.go

```diff
@@ -173,6 +174,25 @@ func ImportRegistryConfig() error {
 	} else {
 		log.Debug("hostname not found, not setting")
 	}
+	if val, _, err = k.GetStringValue("hostname"); err == nil {
+		config.Datadog.Set("hostname", val)
+		log.Debugf("Setting hostname %s", val)
+	} else {
+		log.Debug("hostname not found, not setting")
+	}
+	if val, _, err = k.GetStringValue("cmd_port"); err == nil && len(val) > 0 {
+		cmdPortInt, err := strconv.Atoi(val)
+		if err != nil {
+			log.Warnf("Not setting api port, invalid configuration ")
```

Comment:
Could we display the invalid numnber ?
```golang
log.Warnf("Not setting api port, 'cmp_port' is not a number: %s", val)
```

Also should we set this as an Error instead of a Warning ?

Reply:
We are displaying the invalid number `... %s", val`.
As for Error/Warning... imo it's a warning, because the agent's still going to attempt to start, just on the default port.